pkg/sparrow/run.go

```diff
@@ -137,118 +140,89 @@ func (s *Sparrow) Run(ctx context.Context) error {
-				log.ErrorContext(ctx, "Failed to reset config for check, check will run with last applies config", "error", err)
+				logger.FromContext(ctx).ErrorContext(ctx, "Failed to register check", "error", err)
 			}
-			continue
 		}
-
-		// Check is a new Check and needs to be registered
-		s.registerCheck(ctx, name, c)
+		return
 	}
 
-	for existingCheckName, existingCheck := range s.checks {
-		if _, ok := s.cfg.Checks[existingCheckName]; ok {
-			// Check is known check
-			continue
+	// unregister checks that are not in the new config
+	for name, check := range s.checks {
+		if !cfg.HasCheck(name) {
+			s.unregisterCheck(ctx, check)
 		}
-
-		// Check has been removed from config
-		s.unregisterCheck(ctx, existingCheckName, existingCheck)
-	}
-}
-
-// updateCheckTargets updates the targets of a check with the
-// global targets. The targets are merged per default, if found in the
-// passed config.
-func (s *Sparrow) updateCheckTargets(cfg any) any {
-	if cfg == nil {
-		return nil
 	}
 
-	// check if map with targets
-	checkCfg, ok := cfg.(map[string]any)
-	if !ok {
-		return checkCfg
-	}
-	if _, ok = checkCfg["targets"]; !ok {
-		return checkCfg
-	}
-
-	// Check if targets is a slice
-	actuali, ok := checkCfg["targets"].([]any)
-	if !ok {
-		return checkCfg
-	}
-	if len(actuali) == 0 {
-		return checkCfg
-	}
+	// register / update checks that are in the new config
+	for _, c := range fChecks {
+		if _, ok := s.checks[c.Name()]; !ok {
+			err = s.registerCheck(ctx, c)
+			if err != nil {
+				logger.FromContext(ctx).ErrorContext(ctx, "Failed to register check", "error", err)
+			}
+			continue
+		}
 
-	// convert to string slice
-	var actual []string
-	for _, v := range actuali {
-		if _, ok := v.(string); !ok {
-			return checkCfg
+		// existing config
+		err = s.checks[c.Name()].SetConfig(c.GetConfig())
+		if err != nil {
+			logger.FromContext(ctx).ErrorContext(ctx, "Failed to set config for check", "error", err)
 		}
-		actual = append(actual, v.(string))
 	}
-	var urls []string
+}
 
-	if s.tarMan == nil {
-		return checkCfg
+// enrichTargets updates the targets of the sparrow's checks with the
+// global targets. Per default, the two target lists are merged.
+func (s *Sparrow) enrichTargets(cfg runtime.Config) runtime.Config {
+	if cfg.Empty() || s.tarMan == nil {
+		return cfg
 	}
-	gt := s.tarMan.GetTargets()
 
-	// filter out globalTargets that are already in the config and self
-	for _, t := range gt {
-		if slices.Contains(actual, t.Url) {
+	for _, gt := range s.tarMan.GetTargets() {
+		if gt.Url == fmt.Sprintf("https://%s", s.cfg.SparrowName) {
 			continue
 		}
-		if t.Url == fmt.Sprintf("https://%s", s.cfg.SparrowName) {
-			continue
+		if cfg.HasHealthCheck() && !slices.Contains(cfg.Health.Targets, gt.Url) {
+			cfg.Health.Targets = append(cfg.Health.Targets, gt.Url)
+		}
+		if cfg.HasLatencyCheck() && !slices.Contains(cfg.Latency.Targets, gt.Url) {
+			cfg.Latency.Targets = append(cfg.Latency.Targets, gt.Url)
 		}
```

Comment:
You've consciously didn't enrich the dns ckecks' targets?

Reply:
I didn't get around to do it up until now. Thanks for pointing this out! It'n now implemented and tested.